mobile/lib/shared/ui/immich_image.dart

```diff
@@ -147,64 +109,68 @@ class ImmichImage extends StatelessWidget {
     );
   }
 
-  static AssetEntityImageProvider localImageProvider(
-    Asset asset, {
-    int size = 250,
-  }) =>
-      AssetEntityImageProvider(
-        asset.local!,
-        isOriginal: false,
-        thumbnailSize: ThumbnailSize.square(size),
-      );
-
-  static CachedNetworkImageProvider remoteThumbnailProvider(
-    Asset asset,
-    api.ThumbnailFormat type,
-    Map<String, String> authHeader,
-  ) =>
-      CachedNetworkImageProvider(
-        getThumbnailUrl(asset, type: type),
-        cacheKey: getThumbnailCacheKey(asset, type: type),
-        headers: authHeader,
-      );
-
-  /// TODO: refactor image providers to separate class
-  static CachedNetworkImageProvider remoteThumbnailProviderForId(
-    String assetId, {
-    api.ThumbnailFormat type = api.ThumbnailFormat.WEBP,
-  }) =>
-      CachedNetworkImageProvider(
-        getThumbnailUrlForRemoteId(assetId, type: type),
-        cacheKey: getThumbnailCacheKeyForRemoteId(assetId, type: type),
-        headers: {
-          "x-immich-user-token": Store.get(StoreKey.accessToken),
-        },
-      );
+  static bool useLocal(Asset asset) =>
+      !asset.isRemote ||
+      asset.isLocal && !Store.get(StoreKey.preferRemoteImage, false);
 
+  /// A helper function to use the correct image loader based on
+  /// whether the asset should be local or remote
   /// Precaches this asset for instant load the next time it is shown
-  static Future<void> precacheAsset(
+  static Future<void> precacheAssetImageProvider(
     Asset asset,
     BuildContext context, {
-    type = api.ThumbnailFormat.WEBP,
-    size = 250,
+    api.ThumbnailFormat type = api.ThumbnailFormat.WEBP,
+    int size = 250,
+    ImageErrorListener? onError,
   }) {
     if (useLocal(asset)) {
-      // Precache the local image
       return precacheImage(
-        localImageProvider(asset, size: size),
+        ImmichLocalImageProvider(asset: asset),
         context,
+        onError: onError,
       );
     } else {
-      final accessToken = Store.get(StoreKey.accessToken);
-      // Precache the remote image since we are not using local images
       return precacheImage(
-        remoteThumbnailProvider(asset, type, {"x-immich-user-token": accessToken}),
+        ImmichRemoteImageProvider(assetId: asset.remoteId!),
```

Comment:
`isThumbnail` has to be set after checking if the thumbnail format is WEBP. Alternatively, should we remove the type `param` and replace it with `isThumbnail` as a bool param instead?